File: frontend/src/types/invoice.ts
export type InvoiceStatus = 
  | 'generated'   // Đã tạo
  | 'sent'        // Đã gửi email
  | 'error'       // Lỗi trong quá trình tạo/gửi

export interface InvoiceDto {
  id: number
  invoiceNumber: string
  orderId: number
  customerEmail: string
  customerName: string
  customerPhone?: string
  subtotal: number
  shippingFee: number
  discount: number
  total: number
  status: InvoiceStatus
  filePath?: string
  fileName?: string
  createdAt: string
  updatedAt: string
  sentAt?: string
  errorMessage?: string
}

export interface CreateInvoiceDto {
  orderId: number
  customerEmail: string
  customerName: string
  customerPhone?: string
}

export interface InvoiceGenerationRequest {
  orderId: number
  sendEmail?: boolean
}

export interface InvoiceFilters {
  status?: InvoiceStatus
  search?: string // Tìm theo invoice number, customer name, email
  dateFrom?: string
  dateTo?: string
  page?: number
  pageSize?: number
}

export interface InvoiceListResponse {
  data: InvoiceDto[]
  total: number
  page: number
  pageSize: number
  totalPages: number
}

export interface InvoiceDownloadResponse {
  fileName: string
  content: Blob
}

File: backend/Models/Invoice.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public enum InvoiceStatus
    {
        Generated = 1,    // Đã tạo
        Sent = 2,        // Đã gửi email
        Failed = 3       // Gửi thất bại
    }

    public class Invoice
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(20)]
        public string InvoiceNumber { get; set; } = string.Empty;

        [Required]
        public int OrderId { get; set; }

        [Required]
        [StringLength(255)]
        public string CustomerEmail { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string CustomerName { get; set; } = string.Empty;

        [StringLength(15)]
        public string? CustomerPhone { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal Subtotal { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal ShippingFee { get; set; } = 0;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Discount { get; set; } = 0;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; }

        public InvoiceStatus Status { get; set; } = InvoiceStatus.Generated;

        [StringLength(500)]
        public string? FilePath { get; set; }

        [StringLength(100)]
        public string? FileName { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? SentAt { get; set; }

        [StringLength(500)]
        public string? ErrorMessage { get; set; }

        // Navigation properties
        [ForeignKey("OrderId")]
        public virtual Order Order { get; set; } = null!;
    }
}